src/signer.ts

```diff
@@ -137,7 +143,7 @@ export class URLSigner {
     if (cfg.cname) {
       customHost = cfg.cname;
     } else if (isVirtualHostedStyle) {
-      customHost = `https://${this.bucket.name}.storage.googleapis.com`;
+      customHost = `https://${this.bucket.name}.storage.${this.universeDomain}`;
```

Comment:
Not sure I follow the logic here; is there a way to provide something like "localhost:9000" and create a signed URL that way (either through an endpoint you give the client or otherwise)?

Reply:
Yep, `localhost:9000` can be used via the `signingEndpoint` param to be called to generate a signed URL. Note that a part of the V4 flow the custom host (`cname`) is included in the signed body itself:

https://github.com/googleapis/nodejs-storage/blob/0d3e2111df5c3f7570c07b986c7b4f6d560e888a/src/signer.ts#L251-L254

https://github.com/googleapis/nodejs-storage/blob/0d3e2111df5c3f7570c07b986c7b4f6d560e888a/src/signer.ts#L279

https://github.com/googleapis/nodejs-storage/blob/0d3e2111df5c3f7570c07b986c7b4f6d560e888a/src/signer.ts#L303-L310

https://github.com/googleapis/nodejs-storage/blob/0d3e2111df5c3f7570c07b986c7b4f6d560e888a/src/signer.ts#L312-L328